consensus/src/memory_pool.rs

```diff
@@ -18,244 +18,158 @@
 //!
 //! `MemoryPool` keeps a vector of transactions seen by the miner.
 
-use std::sync::atomic::{AtomicUsize, Ordering};
-
-use crate::error::ConsensusError;
-use mpmc_map::MpmcMap;
-use snarkos_storage::Ledger;
-use snarkvm_algorithms::traits::LoadableMerkleParameters;
-use snarkvm_dpc::{BlockHeader, LedgerScheme, Storage, TransactionScheme, Transactions as DPCTransactions};
-use snarkvm_utilities::{
-    bytes::{FromBytes, ToBytes},
-    has_duplicates,
-    to_bytes,
-};
+use crate::{error::ConsensusError, DynLedger};
+use indexmap::{IndexMap, IndexSet};
+use snarkos_storage::{Digest, SerialRecord, SerialTransaction};
+use snarkvm_dpc::BlockHeader;
+use snarkvm_utilities::has_duplicates;
 
 /// Stores a transaction and it's size in the memory pool.
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
-pub struct Entry<T: TransactionScheme> {
-    pub size_in_bytes: usize,
-    pub transaction: T,
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct Entry {
+    size_in_bytes: usize,
+    transaction: SerialTransaction,
 }
 
 /// Stores transactions received by the server.
 /// Transaction entries will eventually be fetched by the miner and assembled into blocks.
-#[derive(Debug)]
-pub struct MemoryPool<T: TransactionScheme + Send + Sync + 'static> {
+#[derive(Debug, Clone)]
```

Comment:
A deep clone for all these collections within is something we should avoid; what's the rationale?

Reply:
there isn't one -- im not sure why i added it. removed it.